migration/v0.18.x-v0.19.x/curator-retention.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+: "${CK8S_CONFIG_PATH:?Missing CK8S_CONFIG_PATH}"
+
+retention=$(yq r "${CK8S_CONFIG_PATH}"/sc-config.yaml elasticsearch.curator.retention -e 2>&1)
+
+[ "${?}" = "1" ] && echo "No migration needed" && exit 0
+
+echo -e "Retention settings:\n$retention"
+
+patch="
+- command: update
+  path: elasticsearch.curator.retention
+  value:
+"
+
+for alias in other kubeAudit kubernetes authLog; do
+  size=$(yq r "${CK8S_CONFIG_PATH}"/sc-config.yaml "elasticsearch.curator.retention.${alias}SizeGB")
+  age=$(yq r "${CK8S_CONFIG_PATH}"/sc-config.yaml "elasticsearch.curator.retention.${alias}AgeDays")
+  alias_lc=$(echo "${alias}" | awk '{print tolower($0)}')
+
+  # Ensure at least one is set to add pattern
+  [ -n "${size}" ] || [ -n "${age}" ] && patch+="
+      - pattern: ${alias_lc}-*
+  "
+
+  [ -n "${size}" ] && patch+="
+        sizeGB: ${size}
+  "
+
+  [ -n "${age}" ] && patch+="
+        ageDays: ${age}
+  "
+done
```

Comment:
This will still be problematic since `elasticsearch.curator.retention` is an array, and when the config is merging it will be overwriting arrays.
So if only a single property is set, say `otherSizeGB`, then the whole list needs to copied over and set otherwise you would only end up with the entry for `other`.

Reply:
I have updated the script again. Now it should merge with the defaults config.